src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1186,9 +1263,26 @@ public void doLogout(StaplerRequest req, StaplerResponse rsp) throws IOException
         OicSession oicSession = OicSession.getCurrent();
         if (oicSession != null) {
             // session will be invalidated but we still need this data for our redirect.
-            req.setAttribute(ID_TOKEN_REQUEST_ATTRIBUTE, oicSession.getIdToken());
             req.setAttribute(STATE_REQUEST_ATTRIBUTE, oicSession.getState());
         }
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        User user = User.get2(authentication);
+
+        Assert.notNull(user, "User must not be null");
```

Comment:
A null user at logout is not a death sentence - let just ignore that and assume the user is already logged out.

A notice a null id token is not handleld in getPostLogOutUrl2().

Reply:
For "null user", it's to silence spotbugs - this use case is unlikely as it fetches the user from user storage. It is kinda "end of the world" if the user we're trying to log out is not present in the storage.

```
[INFO] --- spotbugs:4.8.3.1:check (spotbugs) @ oic-auth ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[ERROR] Medium: Possible null pointer dereference in org.jenkinsci.plugins.oic.OicSecurityRealm.doLogout(StaplerRequest, StaplerResponse) due to return value of called method [org.jenkinsci.plugins.oic.OicSecurityRealm, org.jenkinsci.plugins.oic.OicSecurityRealm] Dereferenced at OicSecurityRealm.java:[line 1271]Known null at OicSecurityRealm.java:[line 1270] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
```

For ID Token - check has been added and tests have been updated.